### PR TITLE
01594: hip-18 token update

### LINKED ISSUE #1594: Refactor TokenUpdate w/ support for custom fee schedules
**Summary**
The current `TokenUpdateTransitionLogic` backed by a `HederaTokenStore` delegate should be refactored to use the `TypedTokenStore` and `AccountStore`.

While doing this, add support for loading HIP-18 custom fee schedules.

### PR DESCRIPTION
**Pull request by qnswirlds**
**Related issue(s)**:
Closes #1594
More scenarios for #1653, #1654

**Summary of the change**:
- Modify HederaTokenStore to support TokenUpdate with custom fees
- Fix TokenInfo by disallowing setting custom fees and custom fees mutability (`canUpdateWithAdminKey` flag) separately. Also make the setters of those two fields private, just for making a copy of a MerkleToken
- Tokens created without custom fees are not allowed to add custom fees later (custom fees are empty and immutable)
- Add/Enhance unit tests and EET's to allow custom fees settings and validations

**External impacts**:
None.

**Applicable documentation**
None.

### CODE REVIEW

File: hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java [hunk 1]
```diff
@@ -1754,22 +1749,33 @@ void rejectsNegativeFractionInFractionalFee() {
...
-		var result = subject.createProvisionally(req, sponsor, CONSENSUS_NOW);
+		final var result = subject.createProvisionally(req, sponsor, CONSENSUS_NOW);
 
 		// expect:
 		assertEquals(CUSTOM_FEE_MUST_BE_POSITIVE, result.getStatus());
 		assertTrue(result.getCreated().isEmpty());
 	}
 
+	@Test
+	void rejectsZeroAmountInFixedFeeOnTokenUpdate() {
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java:1763]: nit : `rejectsNonPositiveAmountInFixedFeeOnTokenUpdate` would be a better name

**Review by anighanta — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!!

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by ljianghedera — APPROVED** [review]
LGTM

**Review by abhishek-hedera — APPROVED** [review]

File: hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java (hunk 1 above)
Comment (qnswirlds) [reply hedera-node/src/test/java/com/hedera/services/store/tokens/HederaTokenStoreTest.java:1763]: I disagree. This scenario is indeed for zero fixed custom fee. The negative fixed fee was tested with TokenCreate already. So with this test added, we have full validations on the fixed fee.

**Pull request merged**